src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -196,6 +208,46 @@ public void ConnectHeaders_SupplyHeadersThenSupplyMore_Both_Tip_PeerId_Maps_Shou
             Assert.True(peerTipsByPeerIdBefore.Values.Count == peerTipsByPeerIdAfter.Values.Count);
         }
 
+        /// <summary>
+        /// Issue 5 @ Create chained header tree component #1321
+        /// 2 peers should supply headers - one of the peers creates a fork.
+        /// Make sure that ChainedBlock is not created more than once for the same header.
+        /// Check that next pointers were created correctly.
+        /// </summary>
+        [Fact]
+        public void ConnectHeaders_HeadersFromTwoPeersWithFork_ShouldCreateBlocksForNewHeaders()
+        {
+            // Setup
+            var ctx = new TestContext();
+            ChainedHeaderTree cht = ctx.CreateChainedHeaderTree();
+            ChainedHeader chainTip = ctx.ExtendAChain(10);
+            cht.Initialize(chainTip, true);
+            ctx.ChainStateMock.Setup(s => s.ConsensusTip).Returns(chainTip);
+
+            // Default peer is on tip
+            Assert.True(cht.GetPeerTipsByPeerId()[ChainedHeaderTree.LocalPeerId].Equals(chainTip.HashBlock));
+
+            // Peer 1 forks two blocks back
+            BlockHeader forkedBlock = ctx.GetNewBlock(chainTip.Previous.Previous).Header;
+
+            cht.ConnectNewHeaders(1, new List<BlockHeader>() { forkedBlock });
+
+            uint256 peerOneTipHash = cht.GetPeerTipsByPeerId()[1];
+            uint256 hashOfForkedBlock = forkedBlock.GetHash();
+
+            ChainedHeader chainedHeaderOfForkedBlock = cht.GetChainedHeadersByHash()[hashOfForkedBlock];
+            ChainedHeader fork = chainTip.FindFork(chainedHeaderOfForkedBlock);
+
+            fork.Should().NotBeNull();
+
+            peerOneTipHash.Should().Be(hashOfForkedBlock, "Peer should be on the fork but wasn't");
+
+            // Next pointers were created correctly
+            Assert.Equal(2, fork.Next.Count);   // main chain + 1 branch
```

Comment:
can this be represented without the comment, by giving the magic number 2 that meaning